websocket.el

```diff
@@ -693,14 +693,27 @@ to the websocket protocol.
                           (port (if (= 0 (url-port url-struct))
                                     (if (eq type 'tls) 443 80)
                                   (url-port url-struct)))
-                          (host (url-host url-struct)))
+                          (host (url-host url-struct))
+                          (url-as-http (let ((url-as-http (copy-sequence url-struct)))
+                                         (setf (url-type url-as-http) (if (eq type 'plain) "http" "https"))
+                                         url-as-http))
+                          (proxy (url-generic-parse-url (url-find-proxy-for-url url-as-http (url-host url-as-http)))))
                      (if (eq type 'plain)
                          (make-network-process :name name :buffer nil :host host
                                                :service port :nowait nowait)
+                       (if proxy
+                           (let ((plain-conn (make-network-process
+                                              :name name :buffer nil :host (url-host proxy)
+                                              :service (url-port proxy))))
+                             (let ((url-http-after-change-function)
+                                   (url-current-object url-as-http))
+                               (url-https-proxy-connect plain-conn))
+                             (sleep-for 0.5)
+                             (gnutls-negotiate :process plain-conn :hostname host))
```

Comment:
Is this all a standard sequence of actions for https proxy connections?  How would we be sure this is correct?

Reply:
As far as I know it's pretty standard - I figured this out mostly by reading url.el.

We certainly could add a test for this. I looked briefly at https://tinyproxy.github.io/ and it looks like it should support proxying websocket connections. I just didn't get around to it because I wanted to figure out quickly if I could get this working against the internal corporate proxy that I wanted to access.